app/models/order.rb

```diff
@@ -0,0 +1,17 @@
+class Order < ApplicationRecord
+  belongs_to :customer
+  validates_associated :customer
+  validates_presence_of :customer
+  validates :customer_id, presence: true
```

Comment:
These 2 validations perform the same action which is validation of the presence of a **:customer**. I suggest removing either one to avoid any **duplicate** errors.